google/cloud/forseti/scanner/audit/kms_rules_engine.py

```diff
@@ -0,0 +1,469 @@
+                    resource_rule.find_violations(key))
+
+        LOGGER.debug('Returning violations: %r', violations)
+        return violations
+
+
+class ResourceRules(object):
+    """An association of a resource to rules."""
+
+    def __init__(self,
+                 resource=None,
+                 rules=None):
+        """Initialize.
+
+        Args:
+            resource (Resource): The resource to associate with the rule.
+            rules (set): rules to associate with the resource.
+        """
+        if not isinstance(rules, set):
+            rules = set([])
+        self.resource = resource
+        self.rules = rules
+
+    def find_violations(self, key):
+        """Determine if the policy binding matches this rule's criteria.
+
+        Args:
+            key (CryptoKey): crypto key resource.
+
+        Returns:
+            list: RuleViolation
+        """
+        violations = []
+        for rule in self.rules:
+            rule_violations = rule.find_violations(key)
+            if rule_violations:
+                violations.extend(rule_violations)
+        return violations
+
+    def __eq__(self, other):
+        """Compare == with another object.
+
+        Args:
+            other (ResourceRules): object to compare with
+
+        Returns:
+            int: comparison result
+        """
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return (self.resource == other.resource and
+                self.rules == other.rules)
+
+    def __ne__(self, other):
+        """Compare != with another object.
+
+        Args:
+            other (object): object to compare with
+
+        Returns:
+            int: comparison result
+        """
+        return not self == other
+
+    def __repr__(self):
+        """String representation of this node.
+
+        Returns:
+            str: debug string
+        """
+        return 'KMSResourceRules<resource={}, rules={}>'.format(
+            self.resource, self.rules)
+
+
+class Rule(object):
+    """Rule properties from the rule definition file, also finds violations."""
+
+    def __init__(self, rule_name, rule_index, rule):
+        """Initialize.
+
+        Args:
+            rule_name (str): Name of the loaded rule.
+            rule_index (int): The index of the rule from the rule definitions.
+            rule (dict): The rule definition from the file.
+        """
+        self.rule_name = rule_name
+        self.rule_index = rule_index
+        self.rule = rule
+
+    def is_key_rotated(self, creation_time, scan_time):
+        """Check if the key has been rotated within the time speciifed in the
+         policy.
+
+        Args:
+            creation_time (datetime): The time at which the primary version of
+            the key was created.
+            scan_time (datetime): Snapshot timestamp.
+
+        Returns:
+            bool: Returns true if key was not rotated within the time specified
```

Comment:
You should either change the method to reflect the name of the method, is_key_rotated should be true if the key is rotated.